### PR TITLE
[ERM] Unlink linked applications

### PR DESCRIPTION
**Pull request by gherceg**
## Summary

[JIRA Ticket](https://dimagi-dev.atlassian.net/browse/SS-106)

Keeping this PR low risk by not adding this behavior into the actual delete flow yet.

When a domain link is deleted, any LinkedApplication in the downstream domain should be converted to an Application. This builds on existing copy behavior and deletes the LinkedApplication doc once converted as well.

This could be more sophisticated and only unlink applications with an upstream_id correlating to the upstream domain, but since we do not support a domain having more than 1 upstream domain, it does not seem worth implementing that.

Review by commit.

## Feature Flag

`LINKED_DOMAINS`
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Wrote tests along the way
### QA Plan

No QA necessary yet.
### Safety story

Nothing calls this code besides tests at the moment.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/linked_domain/applications.py [hunk 1]
```diff
@@ -129,6 +139,7 @@ def unlink_app(linked_app):
     app_copy.save()
 
     # remove linked app
-    linked_app.delete()
+    full_linked_app = get_app(linked_app.domain, linked_app._id)
```
Comment (orangejenny) [inline corehq/apps/linked_domain/applications.py:142]: We use full apps much more often than briefs. Current naming conventions are that `app` or `linked_app` will refer to an actual app, and then brief apps use `brief` in their identifier.

File: corehq/apps/linked_domain/applications.py [hunk 2]
```diff
@@ -105,3 +106,19 @@ def link_app(linked_app, master_domain, master_id, remote_details=None):
...
+
+
+def unlink_app(linked_app):
+    if not is_linked_app(linked_app):
+        return None
+
+    from corehq.apps.app_manager.models import import_app
+    extra_properties = {'name': linked_app.name}
+    app_copy = import_app(linked_app._id, linked_app.domain, extra_properties)
+    app_copy = app_copy.convert_to_application()
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/linked_domain/applications.py:138]: It was a decision made from ignorance 😄 Thinking back on it, I was under the impression I had to copy the application and then convert the copy, but I see no reason why I can't just convert the existing application like you said. And given that the app id is referenced elsewhere that seems like a much better solution.

File: corehq/apps/linked_domain/applications.py (hunk 1 above)
Comment (gherceg) [reply corehq/apps/linked_domain/applications.py:142]: Good to know.

File: corehq/apps/linked_domain/applications.py [hunk 3]
```diff
@@ -91,6 +91,16 @@ def create_linked_app(master_domain, master_id, target_domain, target_name, remo
     return link_app(linked_app, master_domain, master_id, remote_details)
 
 
+def get_linked_apps_for_domain(domain):
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/linked_domain/applications.py:94]: Due to a change in `unlink_apps_for_domain` I modified this to fetch full apps. If there are performance concerns, I can change it back to fetch brief apps and convert to full apps when needed.

File: corehq/apps/linked_domain/tests/test_application_util_methods.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+from corehq.apps.app_manager.models import LinkedApplication
+from corehq.apps.linked_domain.applications import get_linked_apps_for_domain
+
+
+class ApplicationUtilTests(TestCase):
+
+    def test_get_linked_apps_for_domain(self):
+        linked_app1 = LinkedApplication.new_app('domain', 'One')
+        linked_app2 = LinkedApplication.new_app('domain', 'Two')
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/linked_domain/tests/test_application_util_methods.py:11]: Added an additional test case

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/linked_domain/applications.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/linked_domain/applications.py:94]: Sounds good, avoiding optimizing prematurely is good, especially for an infrequent workflow.

**Pull request merged**